pyproject.toml

```diff
@@ -0,0 +1,162 @@
+[project]
+name = "espnet"
+dynamic = ["version", "readme"]
+authors = [
+    {name="Shinji Watanabe", email="shinjiw@ieee.org"}
+]
+description = "ESPnet: end-to-end speech processing toolkit"
+classifiers = [
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "Operating System :: POSIX :: Linux",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+requires-python = ">=3.10"
+license = { "text" = "Apache Software License" }
+dependencies = [
+    "setuptools>=38.5.1,<74.0.0",
+    "configargparse>=1.2.1",
+    "packaging",
+    "typeguard",
+    "humanfriendly",
+    "scipy>=1.4.1",
+    "numpy<1.24",
+    "PyYAML>=5.1.2",
+    "soundfile>=0.10.2",
+    "torch>=1.11.0",
+    "torchaudio>=1.11.0",
+    "soundfile>=0.10.2",
+    "kaldiio>=2.18.0",
+    "sentencepiece==0.2.0",
+    "librosa>0.9.2",
+    "lightning",
+    "torch_complex",
+    "tensorboard",
+    "einops",
+    "lhotse",
+    "espnet_model_zoo",
+    "matplotlib",
+]
+
+[project.urls]
+Homepage = "http://github.com/espnet/espnet"
+
+[build-system]
+requires = ["setuptools>=38.5.1,<74.0.0", "wheel", "setuptools-scm>=8.0"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools.dynamic]
+version = {attr = "espnet.__version__"}
+readme = {file = ["README.md"], content-type = "text/markdown"}
+
+[tool.setuptools.packages.find]
+include = ["espnet*"]
+
+[tool.setuptools.package-data]
+espnet = ["version.txt"]
+
+[project.optional-dependencies]
+
+espnet-3 = [
```

Comment:
I cannot understand the position of this.
How is it used?

Reply:
It will be used in espnet-3, but maybe we should add this dependency in future PR.